volatility3/framework/plugins/windows/skeleton_key_check.py

```diff
@@ -0,0 +1,561 @@
+
+    def _get_cryptdll_types(self, context: interfaces.context.ContextInterface, 
+                                  config, 
+                                  config_path: str, 
+                                  proc_layer_name: str, 
+                                  cryptdll_base: int):
+        """
+        Builds a symbol table from the cryptdll types generated after binary analysis
+
+        Args:
+            context: the context to operate upon
+            config: 
+            config_path:
+            proc_layer_name: name of the lsass.exe process layer
+            cryptdll_base: base address of cryptdll.dll inside of lsass.exe
+        """
+        table_mapping = {"nt_symbols":  config["nt_symbols"]}
+
+        cryptdll_symbol_table = intermed.IntermediateSymbolTable.create(context = context, 
+                                                                        config_path = config_path,
+                                                                        sub_path = "windows",
+                                                                        filename = "kerb_ecrypt",
+                                                                        table_mapping = table_mapping)
+
+        return context.module(cryptdll_symbol_table, proc_layer_name, offset = cryptdll_base)
+
+    def _find_and_parse_cryptdll(self, proc_list: Iterable) -> \
+                                Tuple[interfaces.context.ContextInterface, str, int, int]:  
+        """
+        Finds the base address of cryptdll.dll insode of lsass.exe
+
+        Args:
+            proc_list: the process list filtered to just lsass.exe instances
+
+        Returns:
+            A tuple of:
+            lsass_proc: the process object for lsass.exe
+            proc_layer_name: the name of the lsass.exe process layer
+            cryptdll_base: the base address of cryptdll.dll
+            crytpdll_size: the size of the VAD for cryptdll.dll
+        """
+        lsass_proc    = None
+        proc_layer_name = None
+        cryptdll_base = None
+        cryptdll_size  = None
+
+        for proc in proc_list:
+            try:
+                proc_id = proc.UniqueProcessId
+                proc_layer_name = proc.add_process_layer()
+            except exceptions.InvalidAddressException as excp:
+                vollog.debug("Process {}: invalid address {} in layer {}".format(proc_id, excp.invalid_address,
+                                                                                 excp.layer_name))
+                continue
+
+            proc_layer = self.context.layers[proc_layer_name]
+
+            for vad in proc.get_vad_root().traverse():
+                filename = vad.get_file_name()
+                
+                if isinstance(filename, str) and filename.lower().endswith("cryptdll.dll"):
+                    cryptdll_base = vad.get_start()
+                    cryptdll_size = vad.get_end() - cryptdll_base
```

Comment:
Sooooo, if we know we're done here, why not just return at this point?  Going through the breaks seems a little pointless, the `lsass_proc = proc` could be moved up to here, and then just the `return lsass_proc, proc_layer_name, cryptdll_base, cryptdll_size` used?  If it's for style (ie, not returning mid-call) I'd argue that breaks are worse for code comprehension.  You may need a `return None, None, None` at the end anyway, but I think overall it's a better way of letting the user figure out what's going on (than iterating over a proc_list that looks like you immediately break out of after a single iteration)?